Tests/Runtime/AgentDispatcherTest.cs

```diff
@@ -47,80 +57,111 @@ private static DoNothingAgent CreateDoNothingAgent(string name = nameof(DoNothin
             return doNothingAgent;
         }
 
+        private void SetUpDispatcher(AutopilotSettings settings)
+        {
+            var logger = new ConsoleLogger(Debug.unityLogger.logHandler);
+            var randomFactory = new RandomFactory(0);
+
+            _dispatcher = new AgentDispatcher(settings, logger, randomFactory);
+        }
+
         private const string TestScenePath = "Packages/com.dena.anjin/Tests/TestScenes/Buttons.unity";
+        private const string TestScenePath2 = "Packages/com.dena.anjin/Tests/TestScenes/Error.unity";
 
-        private static Scene LoadTestScene()
+        private static async UniTask<Scene> LoadTestSceneAsync(string path, LoadSceneMode mode = LoadSceneMode.Single)
         {
-            return EditorSceneManager.LoadSceneInPlayMode(
-                TestScenePath,
-                new LoadSceneParameters(LoadSceneMode.Single));
+            Scene scene = default;
+#if UNITY_EDITOR
+            scene = EditorSceneManager.LoadSceneInPlayMode(path, new LoadSceneParameters(mode));
+            while (!scene.isLoaded)
+            {
+                await Task.Yield();
+            }
+#endif
+            return scene;
         }
 
         [Test]
-        public void DispatchByScene_DispatchAgentBySceneAgentMaps()
+        public async Task DispatchByScene_DispatchAgentBySceneAgentMaps()
         {
-            const string ActualAgentName = nameof(DoNothingAgent);
-
+            const string AgentName = "Mapped Agent";
             var settings = CreateAutopilotSettings();
             settings.sceneAgentMaps.Add(new SceneAgentMap
             {
-                scenePath = TestScenePath, agent = CreateDoNothingAgent()
+                scenePath = TestScenePath, agent = CreateDoNothingAgent(AgentName)
             });
+            SetUpDispatcher(settings);
 
-            var logger = new ConsoleLogger(Debug.unityLogger.logHandler);
-            var randomFactory = new RandomFactory(0);
-            var dispatcher = new AgentDispatcher(settings, logger, randomFactory);
-            dispatcher.DispatchByScene(LoadTestScene());
+            await LoadTestSceneAsync(TestScenePath);
 
-            var gameObject = GameObject.Find(ActualAgentName);
-            Assert.That(gameObject, Is.Not.Null);
+            var actual = Object.FindObjectsOfType<AsyncDestroyTrigger>().Select(x => x.name);
+            Assert.That(actual, Is.EquivalentTo(new[] { AgentName }));
```

Comment:
@nowsprinting I have a question about [`AsyncDestroyTrigger`](https://cysharp.github.io/UniTask/api/Cysharp.Threading.Tasks.Triggers.AsyncDestroyTrigger.html). I could not find the spec about AsyncDestroyTrigger's name field. Could you provide the source of the spec?

Reply:
See also:
https://github.com/DeNA/Anjin/blob/4bc67ddfd01b4fa6fcf9e2f36ab559fc72a15297/Runtime/AgentDispatcher.cs#L106

Reply:
Ah, OK. I misunderstood. I had thought the name of components is independent from attached GameObject. But it was not correct.

Reply:
I feel that what property tested is hard to understand. Before this patch, this test ensures to add an agent by DispatchByScene, it is the tested property. After this patch, what property ensured by this test...?

Reply:
Before the change, it asserts that there is one or more GameObject with "AgentName".
After the change, it is asserted that there is **only one** GameObject with "AgentName".

The only change to this assertion that actually makes sense is `DispatchByScene_ReActivateScene_NotCreateDuplicateAgents` case.
However, I decided that this assertion (only one Agent) was validating a more rigorous state, so I applied it to this test case as well.

Reply:
I try to refactor to clarify intent. wait.

Reply:
Fixed tests using spy agent (not use `AsyncDestroyTrigger`) in 5cc6137